dashboard/app/controllers/api/v1/pd/forms_controller.rb

```diff
@@ -10,10 +10,7 @@ def create
     form_data_json = form_data_hash ? form_data_hash.to_unsafe_h.to_json.strip_utf8mb4 : {}.to_json
     form.form_data_hash = JSON.parse(form_data_json)
 
-    status = params[:status]
-    if status
-      @application.status = status
-    end
+    form.status = new_status if form.is_a? Pd::Application::TeacherApplication
```

Comment:
Is this fixing a bug? It makes sense but I'm not sure how it fits into the rest of this PR

Reply:
This should only be run for teacher apps because it's not relevant for principal apps (or any other app type that inherits from form).

The previous code was a bit janky (I wrote it so I can say that tehe) because the local `status` variable was only set with teacher apps